### PR TITLE
Search results should use the same view settings as the folder being searched

### PR DESCRIPTION
**Pull request by grozahn**
### The demo (with GIFs) of resolved issues in commit comments.

### CODE REVIEW

**Comment by politas** [comment]
Stop thinking you know better than I do what I want to see! I set a folder to a particular view setting, then I search for items in that folder. I want the view setting to _stay the same_. Don't replace the modified date/time column, because maybe that's the column I want to see!

**Comment by lukefromdc** [comment]
This works, but I agree that if users cannot change these defaults we can't merge it as-is. At MATE we cannot take away choices users have had in the past, as the "coders know better than users" concept is not what we are about.

**Comment by grozahn** [comment]
So, it should get view settings for specific folder, but not from global settings? If it, i can quickly fix that issue.

**Comment by grozahn** [comment]
Search results view works in that way that you want. I set settings only for specific folder, not globally. If default view is "List View", but you set "Icon View" for folder, so it will use them.

**Comment by grozahn** [comment]
That's my mistake because i don't showed that on GIF.

**Comment by grozahn** [comment]
**I removed one commit from request (Location column by default), and here's a demo with more details.**
caja_search

**Comment by grozahn** [comment]
⌛

File: src/caja-window-manage-views.c [hunk 1]
```diff
@@ -1266,7 +1266,18 @@ got_file_info_for_view_selection_callback (CajaFile *file,
         /* Otherwise, use default */
         if (view_id == NULL)
         {
-            view_id = caja_global_preferences_get_default_folder_viewer_preference_as_iid ();
+            char *uri;
+            uri = caja_file_get_uri (file);
```
Comment (sc0w) [inline src/caja-window-manage-views.c]: I think this variable (uri) needs to be freed

**Comment by sc0w** [comment]
@politas @mate-desktop/core-team please test

**Comment by raveit65** [comment]
A Link doesn't belongs in commit header, please use commit body for links.
Can you change that please?

**Comment by raveit65** [comment]
commit header = a short description what the commit does
commit body = further detailed infos and links to bugreports.

**Comment by raveit65** [comment]
This is much better when you browse git history.
```
commit 27ca0ac7e6b73b1c633674d119f03a8e19dcd3fd (HEAD -> grozahn-master)
Author: Andrey Streltsov <grozahn@gmail.com>
Date:   Fri Aug 9 00:02:11 2019 +0300

    Use the same view settings for search results as the folder being searched
    
    Fix issue https://github.com/mate-desktop/caja/issues/704
```

**Comment by raveit65** [comment]
Thanks, merged.
https://github.com/mate-desktop/caja/commit/27ca0ac7e6b73b1c633674d119f03a8e19dcd3fd

**Comment by politas** [comment]
I'm not actually running mate desktop at the moment. The gif above shows the date modified column being replaced with location when searching from a detailed list. Is that the behaviour that has been implemented?

**Comment by grozahn** [comment]
I just didn't removed that code for myself.

**Comment by raveit65** [comment]
which code is removed for confusing people?
@politas 
Please install mate in a VM or what ever and test PR.
Asking other people to match your bounty isn't a solution.
You're are the person which have to say `yes` at bountysource.
This `GO` depending on yourself.
I hope you're satisfied....

**Comment by grozahn** [comment]
When i made pull request, i pushed 2 commits and one of them was commit with code which replaces "Date Modified" column with "Locaion" in search results. Soon i removed that commit.

**Comment by grozahn** [comment]
@politas have you tested this?

**Comment by politas** [comment]
Not yet. Trying now.

**Comment by politas** [comment]
I'm getting a 502 error on https://wiki.mate-desktop.org/dev-doc 
Anyone able to tell me the build process for Caja?

**Comment by politas** [comment]
Sheesh! You don't make it easy to build from source!

`$ ./autogen.sh`
`You need to install mate-common from the MATE git`

Wow. So much useful information.

**Comment by lukefromdc** [comment]
I can confirm the "bad gateway" error on https://wiki.mate-desktop.org/dev-doc . Mate-common is simple to install. I have all of mate (starting with that, then mate-desktop) built from source, so building any part of MATE works unless there is a bad commit breaking a build.

**Comment by raveit65** [comment]
@politas 
You didn't wrote which distro are you using in report?
https://github.com/mate-desktop/caja/issues/704#issue-198852350
For fedora i can send you a pre-compiled RPM.
@flexiondotorg 
 https://wiki.mate-desktop.org is down. Can you please take a look at it?

**Comment by politas** [comment]
I am using Ubuntu Mate 18.04 for testing. I have managed to get mate-common to compile and install through trial and error, but then trying autogen.sh on caja repo, I get a list of packages requirements not met that I don't seem to be able to meet - glib-2.0, mate-desktop-2.0, etc. 

I am still just guessing about the build process. Can anyone post a quick list of the commands used for building from source?

**Comment by politas** [comment]
I'm upgrading my VM to Ubuntu mate 19.04. Hopefully that will let me get the right package versions.

**Comment by grozahn** [comment]
How to build deb package on Ubuntu, is there script like PKGBUILD in Arch?

**Comment by grozahn** [comment]
Tried to build in Ubuntu 18.04 with all required dependencies, but also got trouble.

**Comment by grozahn** [comment]
In Ubuntu 18.04 old version of glib (2.56.4), which requires 2.58.1 for build

**Comment by sc0w** [comment]
@politas you have here `.deb` packages with this PR:

[pr1279.deb.packages.zip](https://github.com/mate-desktop/caja/files/3511477/pr1279.deb.packages.zip)

**Comment by politas** [comment]
Right, finally managed to test this out and I'm happy with it. Thanks, @grozahn!

**Pull request closed without merging**